src/hocon_schema.erl

```diff
@@ -360,10 +385,47 @@ find_ref(Schema, Name, Acc, Stack, TStack) ->
             Acc;
         false ->
             Fields0 = fields_and_meta(Schema, Name),
```

Comment:
can we simply drop the hidden fields from here so there is no need to recursively resolve the children fields ?
i.e. do not expose the `hidden or not` state to `hocon_schema_json` and `hocon_schema_example`.

but we need to switch though, so we can choose to either dump a minimal schema (including hidden fields), or a full schema (including hidden fields)

Reply:
also filter `?MODULE:roots/2` result too.
we'll maybe have to introduce a `Opt` or process-dictionary flag, to set `ignore_hidden` to `true`, then drop hidden roots and struct-fields if the flag is set.

Reply:
>  or process-dictionary flag

I confess this was the first thing that came to mind, given the mutually recursive calls of `find_structs` :sweat_smile: